ui/lib/sync/addon/components/sync-header.ts

```diff
@@ -17,12 +18,17 @@ interface Args {
 
 export default class SyncHeaderComponent extends Component<Args> {
   @service declare readonly version: VersionService;
+  @service declare readonly flags: FlagsService;
 
   get badgeText() {
-    return this.version.hasSecretsSync
-      ? ''
-      : this.version.isCommunity
-      ? 'Enterprise feature'
-      : 'Premium feature';
+    const isManaged = this.flags.isManaged;
+    const onLicense = this.version.hasSecretsSync;
+    const isEnterprise = this.version.isEnterprise;
+
+    if (isManaged) return 'Plus feature';
+    if (isEnterprise && !onLicense) return 'Premium feature';
+    if (!isEnterprise) 'Enterprise feature';
+    // no badge for Enterprise clusters with Secrets Sync on their license--the only remaining option.
+    return undefined;
```

Comment:
totally optional, but if this logic is basically the same as that in `components/sidebar/nav/cluster.js` it'd be nice to wrap that little badge in a component OR wrap the badge text in a helper used in both places. again, optional, since copying and pasting this in 2 places isn't so bad.

Reply:
I was on the fence about this, and decided to just copy paste the logic. If it was a unique function we could use elsewhere I'd for sure opt for a helper, but this is very sync only.